src/util/unicode.cpp

```diff
@@ -197,10 +197,10 @@ void utf16_append_code(unsigned int code, bool swap_bytes, std::wstring &result)
     // encode the code in UTF16, possibly swapping bytes.
     code=code-0x10000;
     unsigned int i1=((code>>10) & 0x3ff) | 0xD800;
-    unsigned int a1=(swap_bytes)?do_swap_bytes(static_cast<uint16_t>(i1)):i1;
```

Comment:
How about instead of widening everything to `uint32_t`, narrow `a1` and `a2` to `uint16_t`?